### PR TITLE
[feature](load-refactor) Unfied mysql load use InsertStmt

### LINKED ISSUE #18476: [Feature] load code refactoring
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

# Motivation
Doris now supports many load methods. The implementation code of these methods is intertwined and the attributes overlap. To improve maintainability, we plan to refactor the code.

# Goals
long-term: unify all data loading methods (brokerload, insert, etc) into one
short-term:
1) unify the attribute specification method and parsing method and default values of various loads;
2) reuse single code base as much as possible and delete duplicate code;

# Plan Overview
We plan to use insert stmt as the common backend for all load methods. Various loads retain their original user interface, but will eventually be converted into insert stmt for planning and subsequent execution.

# Task Breakdown
Phase1: make S3/Broker load work on `insert stmt`

1) #18477
2) #18480
3) #18478

Phase2: make all other load method work on `insert stmt`

1) run streamload with insert stmt planner
2) support mysql load
3) support routine load
...

Phase3: fully support all use cases in all load method
1) insert stmt support writing multi-tables in single transaction
2) insert stmt support sequence column v1
...

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by lvshaokang**
# Proposed changes

Issue Number: close #18476 

## Problem summary

1. Implement MysqlLoadStmt as MysqlLoad's unified entry
2. Use unfied's insertstmt to submit load task
3. Optimized UnifiedLoadStmt constructor make it more clearly

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by lvshaokang** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.69 seconds
 stream load tsv:          422 seconds loaded 74807831229 Bytes, about 169 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          60 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230512041810_clickbench_pr_142590.html

**Review by Hastyshell — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/load/loadv2/LoadManagerAdapter.java [hunk 1]
```diff
@@ -31,15 +37,25 @@
...
+            throws UserException, IOException {
         final LoadType loadType = insertStmt.getLoadType();
         final LoadManager loadManager = Env.getCurrentEnv().getLoadManager();
         switch (loadType) {
             case BROKER_LOAD:
                 loadManager.createLoadJobFromStmt(insertStmt);
+                context.getState().setOk();
                 break;
             case MYSQL_LOAD:
                 // TODO: implement
```
Comment (Yukang-Lian) [inline fe/fe-core/src/main/java/org/apache/doris/load/loadv2/LoadManagerAdapter.java:52]: useless code

File: fe/fe-core/src/main/java/org/apache/doris/load/loadv2/MysqlLoadManager.java [hunk 2]
```diff
@@ -148,9 +149,13 @@ public MysqlLoadManager(TokenManager tokenManager) {
 
     public LoadJobRowResult executeMySqlLoadJobFromStmt(ConnectContext context, LoadStmt stmt, String loadId)
             throws IOException, UserException {
+        return executeMySqlLoadJobFromStmt(context, stmt.getDataDescriptions().get(0), loadId);
+    }
+
+    public LoadJobRowResult executeMySqlLoadJobFromStmt(ConnectContext context, DataDescription dataDesc, String loadId)
+            throws IOException, UserException {
         LoadJobRowResult loadResult = new LoadJobRowResult();
         // Mysql data load only have one data desc
```
Comment (Yukang-Lian) [inline fe/fe-core/src/main/java/org/apache/doris/load/loadv2/MysqlLoadManager.java:158]: ditto

File: fe/fe-core/src/main/java/org/apache/doris/analysis/InsertStmt.java [hunk 3]
```diff
@@ -117,7 +117,7 @@ public static class Properties {
     // ---------------------------- for old insert stmt ----------------------------
 
     public boolean isValuesOrConstantSelect() {
-        throw new UnsupportedOperationException("only invoked in NativeInsertStmt");
+        return false;
```
Comment (Yukang-Lian) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/InsertStmt.java:120]: why change this?

Comment (lvshaokang) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/InsertStmt.java:120]: beacuse of the insert stmt as the abstract class, execute to this point[1] will throw exception when i tested. But I think this isn't correct.

[1] https://github.com/apache/doris/blob/master/fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java#L324

File: fe/fe-core/src/main/java/org/apache/doris/analysis/MysqlLoadStmt.java [hunk 4]
```diff
@@ -31,11 +31,23 @@
...
 
+    public MysqlLoadStmt(DataDescription dataDescription, Map<String, String> properties, String comments) {
+        this.label = new LabelName();
+        this.dataDescription = dataDescription;
+        this.properties = properties;
+        if (comments != null) {
+            this.comments = comments;
+        } else {
+            this.comments = "";
+        }
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/MysqlLoadStmt.java:48]: we should handle there in base class.

**Review by dataroaring — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/analysis/MysqlLoadStmt.java [hunk 5]
```diff
@@ -69,7 +81,7 @@ public void analyze(Analyzer analyzer) throws AnalysisException, UserException {
             for (String path : dataDescription.getFilePaths()) {
                 if (Config.mysql_load_server_secure_path.isEmpty()) {
                     throw new AnalysisException("Load local data from fe local is not enabled. If you want to use it,"
-                            + " plz set the `mysql_load_server_secure_path` for FE to be a right path.");
+                        + " plz set the `mysql_load_server_secure_path` for FE to be a right path.");
```
Comment (freemandealer) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/MysqlLoadStmt.java:84]: unnecessary change

**Review by freemandealer — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/analysis/MysqlLoadStmt.java (hunk 5 above)
Comment (lvshaokang) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/MysqlLoadStmt.java:84]: may be checkstyle auto format

File: fe/fe-core/src/main/java/org/apache/doris/load/loadv2/MysqlLoadManager.java [hunk 6]
```diff
@@ -209,6 +214,11 @@ public LoadJobRowResult executeMySqlLoadJobFromStmt(ConnectContext context, Load
         return loadResult;
     }
 
+    public LoadJobRowResult executeMySqlLoadJobFromStmt(ConnectContext context, InsertStmt insertStmt, String loadId)
```
[the remark this answers is not in the document]
Comment (lvshaokang) [reply fe/fe-core/src/main/java/org/apache/doris/load/loadv2/MysqlLoadManager.java:217]: right, only rewrite func for unified`insertstmt`

**Comment by lvshaokang** [comment]
run buildall

**Review by Yukang-Lian — DISMISSED** [review]
LGTM

**Comment by lvshaokang** [comment]
run buildall

**Comment by lvshaokang** [comment]
run p0

**Comment by lvshaokang** [comment]
run buildall

**Comment by lvshaokang** [comment]
run p0

**Comment by lvshaokang** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/analysis/UnifiedLoadStmt.java [hunk 7]
```diff
@@ -55,6 +42,31 @@ public void init() {
...
+        StatementBase proxyStmt;
+        final ConnectContext connectContext = ConnectContext.get();
+        if (connectContext != null && connectContext.getSessionVariable().isEnableUnifiedLoad()) {
+            proxyStmt = new BrokerLoadStmt(label, dataDescriptions, brokerDesc, properties, comment);
+        } else {
+            proxyStmt = new LoadStmt(label, dataDescriptions, brokerDesc, properties, comment);
+        }
+        return new UnifiedLoadStmt(proxyStmt);
+    }
+
```
Comment (CalvinKirs) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/UnifiedLoadStmt.java:69]: Generally LGTM
But it would be better if the code style can be optimized. Similar to this code, we can return directly without passing the else judgment again.

Comment (lvshaokang) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/UnifiedLoadStmt.java]: done

**Review by Hastyshell — APPROVED** [review]
LGTM

**Comment by lvshaokang** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.76 seconds
 stream load tsv:          420 seconds loaded 74807831229 Bytes, about 169 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          58 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          76.9 seconds inserted 10000000 Rows, about 130K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230523075038_clickbench_pr_148793.html

**Comment by lvshaokang** [comment]
run p0

**Comment by lvshaokang** [comment]
run p0

**Comment by lvshaokang** [comment]
run p0

**Review by CalvinKirs — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM
